anchore_engine/services/catalog/catalog_impl.py

```diff
@@ -2196,32 +2191,85 @@ def _delete_image_for_real(userId, image_record, dbsession, image_ids, image_ful
         "manifest_data",
         "parent_manifest_data",
     ]:
-        archiveId = imageDigest
-        logger.debug("DELETEing image from archive {}/{}".format(bucket, archiveId))
-        rc = obj_store.delete(userId, bucket, archiveId)
+        try:
+            logger.debug("DELETEing image from archive %s/%s", bucket, image_digest)
+            rc = obj_store.delete(account_id, bucket, image_digest)
+        except Exception:
+            logger.exception(
+                "Error deleting image from archive %s/%s", bucket, image_digest
+            )
 
     # digest/tag-based archiveId buckets
     for bucket in ["vulnerability_scan"]:
-        for fulltag in image_fulltags:
-            archiveId = "{}/{}".format(imageDigest, fulltag)
-            logger.debug("DELETEing image from archive {}/{}".format(bucket, archiveId))
-            rc = obj_store.delete(userId, bucket, archiveId)
+        for full_tag in full_tags:
+            archive_id = "{}/{}".format(image_digest, full_tag)
+            try:
```

Comment:
The addition of a try statement is not needed as the called function within the obj_store object will raise an exception on failure.